python/cudf/cudf/tests/indexes/test_interval.py

```diff
@@ -188,19 +174,24 @@ def test_interval_range_periods_freq_end(end, freq, periods, closed):
 @pytest.mark.parametrize("end_t", INTERVAL_BOUNDARY_TYPES)
 def test_interval_range_periods_freq_end_dtype(periods_t, freq_t, end_t):
     periods, freq, end = periods_t(2), freq_t(3), end_t(10)
-    freq_val = freq.value if isinstance(freq, cudf.Scalar) else freq
-    end_val = end.value if isinstance(end, cudf.Scalar) else end
-    periods_val = (
-        periods.value if isinstance(periods, cudf.Scalar) else periods
-    )
     pindex = pd.interval_range(
-        end=end_val, freq=freq_val, periods=periods_val, closed="left"
+        end=end, freq=freq, periods=periods, closed="left"
     )
     gindex = cudf.interval_range(
         end=end, freq=freq, periods=periods, closed="left"
     )
+    if gindex.dtype.subtype.kind == "f":
+        gindex = gindex.astype(
+            cudf.IntervalDtype(subtype="float64", closed=gindex.dtype.closed)
+        )
+    elif gindex.dtype.subtype.kind == "i":
+        gindex = gindex.astype(
+            cudf.IntervalDtype(subtype="int64", closed=gindex.dtype.closed)
+        )
```

Comment:
I noticed these checks are used in several places. If its not possible to remove it, can we lower it into `assert_eq` with the comment or a add a helper function?

Reply:
Sure I moved this assertion to its own function in [bd393fc](https://github.com/rapidsai/cudf/pull/17844/commits/bd393fcd257e2117a4c0c948020a9a3294e7b6b9)